### PR TITLE
Erc20 decrypto

### PR DESCRIPTION
**Pull request by fgregoribattista**
first erc20 version with fee

### CODE REVIEW

File: test/token/ERC20/ERC20Decrypto.test.js [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+
+    describe('fee', function () {
+        describe('set address fee', function () {
+            it('emits an address fee change event', async function () {
+                const { logs } = await this.token.setAddressFee(anotherAccount, { from: owner });
+                expectEvent.inLogs(logs, 'AddressFeeChange', {
+                    oldOwnerFee: owner,
+                    newOwnerFee: anotherAccount
+                });
+            });
```
Comment (pmprete) [inline test/token/ERC20/ERC20Decrypto.test.js:31]: make _addressFee public, and check if it's the expected one after using the setAddressFee

File: test/token/ERC20/ERC20Decrypto.test.js [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+
+        });
+        describe('set fee', function () {
+            it('emits an fee change event', async function () {
+                const { logs } = await this.token.setFee(fee, { from: owner });
+                expectEvent.inLogs(logs, 'FeesChange', {
+                    oldFeeBasisPoints: zeroBN,
+                    feeBasisPoints: fee
+                });
+            });
```
Comment (pmprete) [inline test/token/ERC20/ERC20Decrypto.test.js:51]: Also check basisPointsRate has the expected value

File: test/token/ERC20/ERC20Decrypto.test.js [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+            it('reverts when the sender has not fee role', async function () {
+                await expectRevert(this.token.setFee(
+                    fee, { from: anotherAccount }), 'ERC20: must have fee role to setFees',
+                );
+            });
+            it('reverts when fee is mayor than 1000', async function () {
```
Comment (pmprete) [inline test/token/ERC20/ERC20Decrypto.test.js:57]: Explain 1000 is 10%

**Review by pmprete — APPROVED** [review]
Except my comments loks good

File: test/token/ERC20/ERC20Decrypto.test.js (hunk 1 above)
Comment (fgregoribattista) [reply test/token/ERC20/ERC20Decrypto.test.js:31]: Done!! will be applied in the next PR

File: test/token/ERC20/ERC20Decrypto.test.js (hunk 2 above)
Comment (fgregoribattista) [reply test/token/ERC20/ERC20Decrypto.test.js:51]: Done!! will be applied in the next PR

File: test/token/ERC20/ERC20Decrypto.test.js (hunk 3 above)
Comment (fgregoribattista) [reply test/token/ERC20/ERC20Decrypto.test.js:57]: Done!! will be applied in the next PR

**Pull request merged**